### PR TITLE
cilium-operator: don't reconcile non-GAMMA xRoutes without a Cilium-managed Gateway

### LINKED ISSUE #34754: cilium-operator: attempts to reconcile non-Cilium xRoutes
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Version

equal or higher than v1.16.0 and lower than v1.17.0

### What happened?

When enabling the Gateway API functionality in the 1.16.1 helm chart in our k8s clusters, the cilium-operator began reconciling HTTPRoutes owned by the Kong Kubernetes Ingress Controller, which caused both controllers to update the same HTTPRoutes in a tight loop.

### How can we reproduce the issue?

1. Install the Cilium helm chart with Gateway API enabled
2. Install the Kong Kubernetes Ingress Controller chart
3. Create a KIC Gateway and an associated HTTPRoute
4. Observe the KIC and Cilium operators reconciling the HTTPRoute in a tight loop

### Cilium Version

v1.16.1

### Kernel Version

6.1.102

### Kubernetes Version

v1.28.12-eks

### Regression

_No response_

### Sysdump

_No response_

### Relevant log output

_No response_

### Anything else?

_No response_

### Cilium Users Document

- [X] Are you a user of Cilium? Please add yourself to the [Users doc](https://github.com/cilium/cilium/blob/main/USERS.md)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by aetimmes**
Please ensure your pull request adheres to the following guidelines:

- [x] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [ ] All code is covered by unit and/or runtime tests where feasible.
- [x] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [x] If your commit description contains a `Fixes: <commit-id>` tag, then
      please add the commit author[s] as reviewer[s] to this issue.
- [x] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [x] Provide a title or release-note blurb suitable for the release notes.
- [x] Are you a user of Cilium? Please add yourself to the [Users doc](https://github.com/cilium/cilium/blob/main/USERS.md)
- [x] Thanks for contributing!

Fixes: #34754

```release-note
operator: don't reconcile non-GAMMA xRoutes without a Cilium-managed Gateway
```

### CODE REVIEW

**Comment by aetimmes** [comment]
originally wrote this against the v1.16.1 branch and did a copy/paste; need to write this against main

File: USERS.md [hunk 1]
```diff
@@ -639,9 +639,9 @@ Users (Alphabetically)
 
     * N: SeatGeek
       D: SeatGeek uses Cilium as the default CNI for AWS hosted clusters
-      U: CNI, ClusterMesh, Network Policy, Hubble
+      U: CNI, ClusterMesh, Network Policy, Hubble, Gateway API (GAMMA)
       L: https://seatgeek.com
-      Q: @byxorna
+      Q: @byxorna, @aetimmes
```
Comment (sayboras) [inline USERS.md]: Thanks, can you split this change into separate PR to make it faster to review and merge?

File: operator/pkg/gateway-api/httproute.go [hunk 2]
```diff
@@ -146,16 +146,28 @@ func (r *httpRouteReconciler) SetupWithManager(mgr ctrl.Manager) error {
...
+		scopedLog := r.logger.With(logfields.Controller, gateway, logfields.Resource, obj.GetName())
 		hr, ok := obj.(*gatewayv1.HTTPRoute)
 		if !ok {
 			return false
 		}
 
 		for _, parent := range hr.Spec.ParentRefs {
 			if helpers.IsGateway(parent) {
-				return true
+				// check if the Gateway has the correct controller
```
Comment (sayboras) [inline operator/pkg/gateway-api/httproute.go]: do we need to update similar logic for grpcroute and tlsroute ?

Comment (aetimmes) [reply operator/pkg/gateway-api/httproute.go]: likely yes - this logic also isn't currently working in my local setup; will revisit later today

**Comment by aetimmes** [comment]
I discovered that my original approach prevented Routes from being watched, but it didn't prevent them from actually being reconciled, so I moved the logic into the actual reconciler funcs.

This feels like it violates DRY, but I couldn't find a good place to unify this logic that didn't create an import cycle. Open to thoughts here.
